Import statements 

In [57]:
import pandas as pd 
import numpy as np 
import json
from collections import defaultdict
import pickle

setting up dataframes 

importing the hpoa file 

In [58]:
hpoa_df = pd.read_csv('/home/tanay/COMAP/Final_concnecous/dictonary_making/phenotype_new.txt',  sep='\t')


/tmp/ipykernel_28185/2265009699.py:1: DtypeWarning: Columns (6,8,9) have mixed types. Specify dtype option on import or set low_memory=False.
  hpoa_df = pd.read_csv('/home/tanay/COMAP/Final_concnecous/dictonary_making/phenotype_new.txt',  sep='\t')


making of first dataframe array from hpoa file 

In [59]:
dict_making_df1 = hpoa_df[['DatabaseID', 'HPO_ID']]
dict_making_df1_array = dict_making_df1.to_numpy()

importing gene to phenotype file 

In [60]:
gene_to_phenotype_df = pd.read_csv('/home/tanay/COMAP/Final_concnecous/dictonary_making/v081223/genes_to_phenotype.txt', sep='\t', low_memory=False)
gene_to_phenotype_df.columns

Index(['ncbi_gene_id', 'gene_symbol', 'hpo_id', 'hpo_name', 'frequency',
       'disease_id'],
      dtype='object')

making of second dataframe from gene to phenotype file 

In [61]:
dict_making_df2 = gene_to_phenotype_df[['disease_id', 'hpo_id']]
dict_making_df2_array = dict_making_df2.to_numpy()

concatination of both numpy arrays of the dataframe created

In [62]:
dict_making_df_array = np.concatenate((dict_making_df1_array, dict_making_df2_array))

len(dict_making_df_array)

547245

In [63]:
dict_making_df_array

array([['OMIM:619340', 'HP:0011097'],
       ['OMIM:619340', 'HP:0002187'],
       ['OMIM:619340', 'HP:0001518'],
       ...,
       ['OMIM:146550', 'HP:0045075'],
       ['OMIM:146550', 'HP:0001596'],
       ['OMIM:146550', 'HP:0030056']], dtype=object)

test dictonary making - naming convention -- 1 (as suffix)

In [64]:
disease_hpo_dict1 = defaultdict(list)

for DiseaseID, HPO_ID in dict_making_df_array:
    disease_hpo_dict1[DiseaseID].append(HPO_ID)

In [65]:
len(disease_hpo_dict1)

12911

In [66]:
hpo_disease_dict1 = defaultdict(list)

for DiseaseID, HPO_ID in dict_making_df_array:
    hpo_disease_dict1[HPO_ID].append(DiseaseID)

In [67]:
len(hpo_disease_dict1)

10796

importing disease vpk count df for disease set

In [68]:
disease_vpk_count_df = pd.read_excel('/home/tanay/COMAP/Final_concnecous/dictonary_making/v081223/Frozen_consensus_MasterSheet_HPODiseaseWithVPK_V_110923.xlsx')
disease_vpk_count_df.columns

Index(['Disease_ID', 'V_Count', 'P_Count', 'K_Count'], dtype='object')

making disease set

In [69]:
disease_set = set(disease_vpk_count_df['Disease_ID'].to_list())
len(disease_set)

12737

importing hpid vpk labek file for hpid set

In [70]:
hpid_vpk_label_df = pd.read_excel('/home/tanay/COMAP/Final_concnecous/dictonary_making/v081223/Frozen_consensus_MasterSheet_HPOtermsWithVPK_V_110923xlsx.xlsx')
hpid_vpk_label_df.columns

Index(['HP_Terms', 'Description', 'V', 'P', 'K'], dtype='object')

making of hpid set

In [71]:
hpid_set = set(hpid_vpk_label_df['HP_Terms'].to_list())
len(hpid_set)

10610

converting concatinated array to dateframe 

In [72]:
dict_making_df1 = pd.DataFrame(dict_making_df_array, columns=['disease_id', 'hp_id'])
dict_making_df1

disease_id       hp_id
0       OMIM:619340  HP:0011097
1       OMIM:619340  HP:0002187
2       OMIM:619340  HP:0001518
3       OMIM:619340  HP:0032792
4       OMIM:619340  HP:0011451
...             ...         ...
547240  OMIM:146550  HP:0002209
547241  OMIM:146550  HP:0000653
547242  OMIM:146550  HP:0045075
547243  OMIM:146550  HP:0001596
547244  OMIM:146550  HP:0030056

[547245 rows x 2 columns]

taking set intersection of the dataframe columns 

In [73]:
dict_making_df = dict_making_df1.query('disease_id in @disease_set and hp_id in @hpid_set')
dict_making_df

disease_id       hp_id
0       OMIM:619340  HP:0011097
1       OMIM:619340  HP:0002187
2       OMIM:619340  HP:0001518
3       OMIM:619340  HP:0032792
4       OMIM:619340  HP:0011451
...             ...         ...
547240  OMIM:146550  HP:0002209
547241  OMIM:146550  HP:0000653
547242  OMIM:146550  HP:0045075
547243  OMIM:146550  HP:0001596
547244  OMIM:146550  HP:0030056

[541373 rows x 2 columns]

converting the treated intersection dataframe to np array

In [74]:
dict_making_df_np = dict_making_df.to_numpy()
dict_making_df_np

array([['OMIM:619340', 'HP:0011097'],
       ['OMIM:619340', 'HP:0002187'],
       ['OMIM:619340', 'HP:0001518'],
       ...,
       ['OMIM:146550', 'HP:0045075'],
       ['OMIM:146550', 'HP:0001596'],
       ['OMIM:146550', 'HP:0030056']], dtype=object)

making of final dictonary - naming convention -- 2

In [75]:
disease_hpo_dict2 = defaultdict(list)

for DiseaseID, HPO_ID in dict_making_df_np:
    disease_hpo_dict2[DiseaseID].append(HPO_ID)

len(disease_hpo_dict2)

12737

In [76]:
hpo_disease_dict2 = defaultdict(list)

for DiseaseID, HPO_ID in dict_making_df_np:
    hpo_disease_dict2[HPO_ID].append(DiseaseID)

len(hpo_disease_dict2)

10604

checking for difference between teh dictonaries keys and set 


phenotype differences 

In [77]:
hpo_disease_dict2_phenotypes_set = set(hpo_disease_dict2.keys())
len(hpo_disease_dict2_phenotypes_set)

10604

In [78]:
diff_phenotypes  = hpid_set - hpo_disease_dict2_phenotypes_set
diff_phenotypes

{'HP:0001475',
 'HP:0009067',
 'HP:0012828',
 'HP:0025293',
 'HP:0025295',
 'HP:0031040'}

diff phenotypes                                                             
{'HP:0001475',
 'HP:0009067',
 'HP:0012828',
 'HP:0025293',
 'HP:0025295',
 'HP:0031040'}

disease_differences 

In [79]:
disease_hpo_dict2_disease_set = set(disease_hpo_dict2.keys())
len(disease_hpo_dict2_disease_set)

12737

In [80]:
diff_diseases = disease_set - disease_hpo_dict2_disease_set
diff_diseases

set()

diff disease - {'OMIM:209108'}


saving the final dict - dict2

In [81]:
def pickle_save(file_name, dict_name):
    with open(file_name, 'wb') as handle:
        pickle.dump(dict_name, handle, protocol=pickle.HIGHEST_PROTOCOL)

In [82]:
version_date = 'v081223'

In [83]:
# pickle_save('hpo_disease_dict_' + version_date + '.pickle', hpo_disease_dict2)

In [84]:
# pickle_save('disease_hpo_dict_' + version_date + '.pickle', disease_hpo_dict2)

In [86]:
hpo_disease_dict2['HP:0004297']

[]